src/signaler.cpp

```diff
@@ -189,6 +189,7 @@ void zmq::signaler_t::send ()
     unsigned char dummy = 0;
     while (true) {
         int nbytes = ::send (w, (char*) &dummy, sizeof (dummy), 0);
+        wsa_assert (nbytes != SOCKET_ERROR);
         if (unlikely (nbytes == SOCKET_ERROR))
```

Comment:
I am not so sure it won't be. I am watching this in my project, things go pretty well so far. Without this patch, `zmq_assert` below fails very often in my program.